[BUG]: Incorrect missing values in pylibcudf.Table.from_arrow on a sliced, string_view column

**Describe the bug**

When creating a `plc.Table.from_arrow` on a pyarrow Table with a *sliced* `string_view` column, something seems to be off about the validity map:

**Steps/Code to reproduce bug**

```python
import pyarrow as pa, pylibcudf as plc

table = pa.table({"a": pa.array(["a", None], pa.string_view())})
roundtrip = plc.interop.to_arrow(plc.interop.from_arrow(table.slice(1, 2)))
result = roundtrip.columns[0][0].as_py()
assert result is None, result
```

That fails

```
AssertionError: 
```

(not the best message, but `result` is the empty string rather than `None`.

**Expected behavior**

`result` should be `None`.

**Environment overview (please complete the following information)**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**

Maybe the root cause of https://github.com/rapidsai/cudf/issues/19148. Check back after fixing.

--- a/cpp/src/interop/from_arrow_host.cu
+++ b/cpp/src/interop/from_arrow_host.cu
@@ -326,10 +326,10 @@ std::unique_ptr<column> get_column_copy(ArrowSchemaView* schema,
                                         rmm::cuda_stream_view stream,
                                         rmm::device_async_resource_ref mr)
 {
-  CUDF_EXPECTS(
-    input->length <= static_cast<std::int64_t>(std::numeric_limits<cudf::size_type>::max()),
-    "Total number of rows in Arrow column exceeds the column size limit.",
-    std::overflow_error);
+  CUDF_EXPECTS((input->length + input->offset) <=
+                 static_cast<std::int64_t>(std::numeric_limits<cudf::size_type>::max()),
+               "Total number of rows in Arrow column exceeds the column size limit.",
+               std::overflow_error);
 
   return type.id() != type_id::EMPTY
            ? std::move(type_dispatcher(

--- a/cpp/src/interop/from_arrow_host_strings.cu
+++ b/cpp/src/interop/from_arrow_host_strings.cu
@@ -152,8 +152,10 @@ std::unique_ptr<column> from_arrow_stringview(ArrowSchemaView* schema,
   // first copy stringview array to device
   auto items   = view.buffer_views[stringview_vector_idx].data.as_binary_view;
   auto d_items = rmm::device_uvector<ArrowBinaryView>(input->length, stream, mr);
+  // offset should not be > input->length which caller insures is < max size_type
+  auto const offset = static_cast<cudf::size_type>(input->offset);
   CUDF_CUDA_TRY(cudaMemcpyAsync(d_items.data(),
-                                items + input->offset,
+                                items + offset,
                                 input->length * sizeof(ArrowBinaryView),
                                 cudaMemcpyDefault,
                                 stream.value()));
@@ -181,8 +183,10 @@ std::unique_ptr<column> from_arrow_stringview(ArrowSchemaView* schema,
     thrust::counting_iterator<cudf::size_type>(0),
     thrust::counting_iterator<cudf::size_type>(input->length),
     d_indices.begin(),
-    [d_items = d_items.data(), d_ptrs, d_mask] __device__(auto idx) -> string_index_pair {
-      if (d_mask && !cudf::bit_is_set(d_mask, idx)) { return string_index_pair{nullptr, 0}; }
+    [d_items = d_items.data(), d_ptrs, d_mask, offset] __device__(auto idx) -> string_index_pair {
+      if (d_mask && !cudf::bit_is_set(d_mask, idx + offset)) {
+        return string_index_pair{nullptr, 0};
+      }
       auto const& item = d_items[idx];
       auto const size  = static_cast<cudf::size_type>(item.inlined.size);
       auto const data  = (size <= NANOARROW_BINARY_VIEW_INLINE_SIZE)

--- a/cpp/src/interop/from_arrow_host_strings.cu
+++ b/cpp/src/interop/from_arrow_host_strings.cu
@@ -152,7 +152,7 @@ std::unique_ptr<column> from_arrow_stringview(ArrowSchemaView* schema,
   // first copy stringview array to device
   auto items   = view.buffer_views[stringview_vector_idx].data.as_binary_view;
   auto d_items = rmm::device_uvector<ArrowBinaryView>(input->length, stream, mr);
-  // caller insures that input->offset is < max size_type
+  // caller ensures that input->offset is < max size_type
   auto const offset = static_cast<cudf::size_type>(input->offset);
   CUDF_CUDA_TRY(cudaMemcpyAsync(d_items.data(),
                                 items + offset,

--- a/cpp/src/interop/from_arrow_host_strings.cu
+++ b/cpp/src/interop/from_arrow_host_strings.cu
@@ -152,7 +152,7 @@ std::unique_ptr<column> from_arrow_stringview(ArrowSchemaView* schema,
   // first copy stringview array to device
   auto items   = view.buffer_views[stringview_vector_idx].data.as_binary_view;
   auto d_items = rmm::device_uvector<ArrowBinaryView>(input->length, stream, mr);
-  // offset should not be > input->length which caller insures is < max size_type
+  // caller insures that input->offset is < max size_type
   auto const offset = static_cast<cudf::size_type>(input->offset);
   CUDF_CUDA_TRY(cudaMemcpyAsync(d_items.data(),
                                 items + offset,

--- a/cpp/tests/interop/from_arrow_host_test.cpp
+++ b/cpp/tests/interop/from_arrow_host_test.cpp
@@ -750,6 +750,32 @@ TEST_F(FromArrowHostDeviceTest, DictionaryIndicesType)
   CUDF_TEST_EXPECT_TABLES_EQUAL(got_cudf_table->view(), from_struct);
 }
 
+void slice_host_nanoarrow(ArrowArray* arr, int64_t start, int64_t end)
+{
+  auto op = [&](ArrowArray* array) {
+    // slicing only needs to happen at the top level of an array
+    array->offset = start;
+    array->length = end - start;
+    if (array->null_count != 0) {
+      array->null_count =
+        array->length -
+        ArrowBitCountSet(ArrowArrayValidityBitmap(array)->buffer.data, start, end - start);
+    }
+  };
+
+  if (arr->n_children == 0) {
+    op(arr);
+    return;
+  }
+
+  // since we want to simulate a sliced table where the children are sliced,
+  // we slice each individual child of the record batch
+  arr->length = end - start;
+  for (int64_t i = 0; i < arr->n_children; ++i) {
+    op(arr->children[i]);
+  }
+}
+
 TEST_F(FromArrowHostDeviceTest, StringViewType)
 {
   auto data = std::vector<std::string>({"hello",
@@ -796,32 +822,13 @@ TEST_F(FromArrowHostDeviceTest, StringViewType)
 
   auto expected = cudf::test::strings_column_wrapper(data.begin(), data.end(), validity.begin());
   CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(result->view(), expected);
-}
 
-void slice_host_nanoarrow(ArrowArray* arr, int64_t start, int64_t end)
-{
-  auto op = [&](ArrowArray* array) {
-    // slicing only needs to happen at the top level of an array
-    array->offset = start;
-    array->length = end - start;
-    if (array->null_count != 0) {
-      array->null_count =
-        array->length -
-        ArrowBitCountSet(ArrowArrayValidityBitmap(array)->buffer.data, start, end - start);
-    }
-  };
-
-  if (arr->n_children == 0) {
-    op(arr);
-    return;
-  }
-
-  // since we want to simulate a sliced table where the children are sliced,
-  // we slice each individual child of the record batch
-  arr->length = end - start;
-  for (int64_t i = 0; i < arr->n_children; ++i) {
-    op(arr->children[i]);
-  }
+  // test with sliced (and include a null row)
+  slice_host_nanoarrow(&input, 2, 4);
+  auto sliced_result = cudf::from_arrow_column(&schema, &input);
+  auto sliced_expected =
+    cudf::test::strings_column_wrapper(data.begin() + 2, data.begin() + 4, validity.begin() + 2);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(sliced_result->view(), sliced_expected);
 }
 
 struct FromArrowHostDeviceTestSlice

